src/events/broker.py

```diff
@@ -470,6 +476,13 @@ def _init_kraft_mode(self) -> None:
                 # single mode, controller & leader
                 self.charm.state.cluster.update({"controller-password": generated_password})
 
+            bootstrap_data = {
+                "bootstrap-controller": self.charm.state.bootstrap_controller,
+                "bootstrap-unit-id": str(self.charm.state.kraft_unit_id),
+                "bootstrap-replica-id": self.workload.generate_uuid(),
+            }
+            self.charm.state.cluster.update(bootstrap_data)
+
     def _format_storages(self) -> None:
```

Comment:
**todo:** I think we should move all the Kraft specific functions out of `events/broker.py` and in to their own dedicated `events/kraft.py`. Reduces the complexity of this file immensely, and ideally we're only handling core lifecycle events for brokers here, as well as setting of peer-cluster/client data.

Reply:
I was a little afraid to do that because of intermingled broker<>controller logic, but I was wrong, now that it's done, seems much more coherent. Thanks!